src/main/java/com/netgrif/application/engine/workflow/service/DataService.java

```diff
@@ -606,12 +601,28 @@ public SetDataEventOutcome saveFile(String taskId, String fieldId, MultipartFile
         ImmutablePair<Case, FileField> pair = getCaseAndFileField(taskId, fieldId);
         FileField field = pair.getRight();
         Case useCase = pair.getLeft();
+        IStorageService storageService = storageResolverService.resolve(field.getStorageType());
+        try {
+            if (useCase.getDataSet().get(field.getStringId()).getValue() != null && field.getValue().getPath() != null) {
+                storageService.delete(field.getValue().getPath());
+                if (field.getValue().getPreviewPath() != null) {
+                    storageService.delete(field.getValue().getPreviewPath());
+                }
+                useCase.getDataSet().get(field.getStringId()).setValue(null);
+            }
+
+            field.setValue(multipartFile.getOriginalFilename());
```

Comment:
Is it possible that multipartFile is null?

Reply:
multipartFile comes from Controller when is marked as required so I think it cannot be null, maybe i can add check for file size?